File: MyDailyBeat/API/VerveAPI/GroupPrefs.m
//
//  GroupPrefs.m
//  VerveAPI
//
//  Created by Virinchi Balabhadrapatruni on 8/17/15.
//  Copyright (c) 2015 eVerveCorp. All rights reserved.
//

#import "GroupPrefs.h"

@implementation GroupPrefs

- (id) initWithServingURL:(NSString *) servingURL {
    self = [super init];
    if (self) {
        [self loadPictureWithServingURL:servingURL];
    }
    return self;
}

- (void) loadPictureWithServingURL:(NSString *) servingURL {
    dispatch_queue_t queue = dispatch_queue_create("dispatch_queue_t_dialog", NULL);
    dispatch_async(queue, ^{
        if (servingURL != nil) {
            NSURL *imageURL = [[NSURL alloc] initWithString:servingURL];
            NSData *imageData = [[RestAPI getInstance] fetchImageAtRemoteURL:imageURL];
            dispatch_async(dispatch_get_main_queue(), ^{
                // Update the UI
                self.groupPicture = [UIImage imageWithData:imageData];
                
            });
        }
    });
    
}

- (NSDictionary *) groupPictureField {
    return @{FXFormFieldTitle: @"Change Group Picture", FXFormFieldAction: @"saveImage:"};
}

- (NSDictionary *) hobbiesField {
    return @{FXFormFieldTitle: @"Select Hobbies", FXFormFieldOptions: @[@"Books", @"Golf", @"Cars", @"Walking", @"Hiking", @"Wine", @"Woodworking", @"Online Card Games", @"Card Games", @"Online Games", @"Arts & Crafts", @"Prayer", @"Support Groups", @"Shopping", @"Travel", @"Local Field Trips", @"History", @"Sports"], FXFormFieldViewController: @"EVCGroupSettingsHobbiesSelectionTableViewController"};
}

- (NSArray *)extraFields
{
    return @[
             @{FXFormFieldTitle: @"Delete Group", FXFormFieldHeader: @"", FXFormFieldAction: @"deleteGroup:", @"contentView.backgroundColor": [UIColor redColor], @"textLabel.color": [UIColor whiteColor]},
             ];
}

@end

File: MDB_Swift/MyDailyBeat/API/src/GroupPrefs.swift
//  Converted with Swiftify v1.0.6242 - https://objectivec2swift.com/
//
//  GroupPrefs.swift
//  VerveAPI
//
//  Created by Virinchi Balabhadrapatruni on 8/17/15.
//  Copyright (c) 2015 eVerveCorp. All rights reserved.
//
import Foundation
public class GroupPrefs: NSObject {
    public var groupPicture: UIImage!
    public var hobbies: [Int: Bool] = [Int: Bool]()

    public init(servingURL: String = "") {
        super.init()
        //self.loadPicture(withServingURL: servingURL)
    
    }


    func loadPicture(withServingURL servingURL: String) {
        
        DispatchQueue.global().async(execute: {() -> Void in
            let imageURL = URL(string: servingURL)
            let imageData: Data? = RestAPI.getInstance().fetchImage(atRemoteURL: imageURL!)
            DispatchQueue.main.async(execute: {() -> Void in
                // Update the UI
                self.groupPicture = UIImage(data: imageData!)
            })
        })
    }
    
    public func toggle(key: Int, value: Bool) {
        let values: [Bool] = hobbies.values.filter({ (val) -> Bool in
            return val
        })
        if value && values.count < 3 {
            self.hobbies[key] = value
        } else if !value {
            self.hobbies[key] = value
        }
    }
    
}
